var/spack/repos/builtin/packages/hpx/package.py

```diff
@@ -35,7 +36,12 @@ class Hpx(CMakePackage, CudaPackage):
 
     variant('networking', default=True,
             description='Support for networking and multi=node runs')
+    variant('mpi', default=False,
+            description='Enable the MPI parcelport (requires the networking'
+            ' variant to be on)')
+
```

Comment:
Maybe add `conflicts('+mpi', when='~networking')`.

Reply:
Thanks, that seems to be exactly what I was looking for.